js/style/style_layer.js

```diff
@@ -38,6 +38,8 @@ StyleLayer.prototype = {
                 }
                 this.layout['symbol-avoid-edges'] = true;
             }
+
+            this._resolvedLayout = new StyleDeclarationSet('layout', this.type, this.layout, this._constants || {});
```

Comment:
`_resolvedLayout` is a set of StyleDeclarations that can be evaluated for individual zooms.
`layout` is just the set of raw style properties with fallbacks.

Back when `layout` properties weren't ever functions they never needed to be evaluated. We should be evaluating layout properties on the main thread too now that they can be functions. New issue: https://github.com/mapbox/mapbox-gl-js/issues/1421